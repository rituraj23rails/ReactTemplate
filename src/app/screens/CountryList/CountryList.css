.countryListMainContainer {
    height: 100vh;
}

.countryContainer {
    display: flex;
    flex-direction: column;
}

.searchedCountryText {
    background-color: #000;
    color: #fff;
    height: 80px;
    display: flex;
    justify-content: center;
    align-items: center;
    position: fixed;
    top: 0;
    width: 100%;
    font-size: 25px;
}

.countryList {
    margin-top: 100px;
    height: 600px;
    overflow: scroll;
}

.countryCell {
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    border: 2px solid #000;
    width: 600px;
    padding: 10px 30px;
    margin: 0px auto 20px auto;
}

.fieldValue {
    font-size: 15px;
    font-weight: bold;
}

.flagImage {
    width: 150px;
    height: 100px;
}

@media screen and (max-width: 820px) {
    .searchedCountryText {
        font-size: 22px;
    }

    .countryList {
        margin-top: 100px;
        height: 100%;
        overflow: scroll;
    }

    .countryCell {
        width: 400px;
    }
}

@media screen and (max-width: 480px) {
    .searchedCountryText {
        font-size: 20px;
    }

    .countryList {
        margin-top: 100px;
        height: 100%;
        overflow: scroll;
    }

    .countryCell {
        width: 300px;
    }
}