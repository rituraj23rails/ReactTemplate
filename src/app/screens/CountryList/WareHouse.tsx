import React, { useState, useEffect } from 'react'
import { Select, MenuItem, Checkbox, Button } from '@mui/material';
import DropdownMenu from 'react-bootstrap/esm';

export let handleChange: any;
export let onAccessSelectionChange: any;

const WareHouse = () => {
    const [checkedData, setCheckedData] = useState<any>([]);
    const [checkBoxData, setCheckBoxData] = useState<any>({});
    const [userSelected, setUserSelected] = useState<any>('Employee');
    const data: any = [
        {
            key: '1',
            displayName: 'Employee',
            checkboxObject: [
                { label: 'Upload Site', value: false },
                { label: 'Delete Site', value: false },
                { label: 'View actvity', value: false }
            ]
        },
        {
            key: '2',
            displayName: 'Manager',
            checkboxObject: [
                { label: 'Upload Site', value: false },
                { label: 'Delete Site', value: false },
                { label: 'View actvity', value: false }
            ]
        },
        {
            key: '3',
            displayName: 'Driver',
            checkboxObject: [
                { label: 'Upload Site', value: false },
                { label: 'Delete Site', value: false },
                { label: 'View actvity', value: false }
            ]
        },
        {
            key: '4',
            displayName: 'Operator',
            checkboxObject: [
                { label: 'Upload Site', value: false },
                { label: 'Delete Site', value: false },
                { label: 'View actvity', value: false }
            ]
        },
        {
            key: '5',
            displayName: 'Leader',
            checkboxObject: [
                { label: 'Upload Site', value: false },
                { label: 'Delete Site', value: false },
                { label: 'View actvity', value: false }
            ]
        }
    ];

    onAccessSelectionChange = (event: any) => {
        setUserSelected(event.target.value);
        const checkboxDataIndex = checkedData.findIndex(
            (checkedItem: any) => String(checkedItem.displayName) === String(event.target.value)
        );
        setCheckBoxData(checkedData[checkboxDataIndex]);
    };

    useEffect(() => {
        setCheckedData(data);
        setCheckBoxData(data[0]);
    }, []);

    handleChange = (event: any, item: any) => {
        const tempCheckedBoxData = [...checkBoxData?.checkboxObject];
        const checkboxDataIndex = tempCheckedBoxData?.findIndex(
            (checkedItem: any) => checkedItem.label === item?.label
        );
        tempCheckedBoxData[checkboxDataIndex].value = !tempCheckedBoxData[checkboxDataIndex].value;
        const tempItem = { ...checkBoxData };
        tempItem.checkboxObject = tempCheckedBoxData;
        setCheckBoxData(tempItem);
        const tempCheckedData = checkedData;
        const checkedDataIndex = checkedData?.findIndex(
            (checkedItem: any) => checkedItem.displayName === checkBoxData.displayName
        );
        tempCheckedData[checkedDataIndex] = tempItem;
        setCheckedData(tempCheckedData);
    };

    return (
        <div>
            <Select
                labelId="demo-simple-select-label"
                id="demo-simple-select"
                value={userSelected}
                label="Display"
                onChange={onAccessSelectionChange}
            >
                {checkedData?.map((item: any) => {
                    return (
                        <MenuItem value={item.displayName}>{item.displayName}</MenuItem>
                    )
                })}
            </Select>

            {checkBoxData?.checkboxObject?.map((item: any) => {
                return (
                    <>
                        <Checkbox
                            color="primary"
                            inputProps={{ 'aria-label': 'uncontrolled-checkbox' }}
                            onChange={(e) => handleChange(e, item)}
                            value={item.value}
                            name={item.label}
                            checked={item.value}
                        />
                        <label>{item.label}</label>
                    </>
                );
            })}
        </div>
    )
}

export default WareHouse
