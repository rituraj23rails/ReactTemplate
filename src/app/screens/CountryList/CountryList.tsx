import React, { Component, useRef } from 'react';
import './CountryList.css';
import WithRouter from '../../../WithRouter';

export interface Props {
    navigate: any;
    location: any;
    id: string;
}

interface S {
    countryData: any;
}

interface SS {
    id: any;
}

class CountryList extends Component<Props, S, SS> {
    carousel: any = null;
    constructor(props: any) {
        super(props);
        this.state = {
            countryData: this.props.location.state.countryList,
        };
    };

    componentDidMount = async () => {
        console.log('@@@ Params =========', this.props.location.state.countryList)
    }

    componentWillUnmount = () => {

    }

    handleChangeTextInput = (event: any) => {
        event.preventDefault();
    }

    onPressCountryDetails = (item: any) => {
        this.props.navigate('/countrydetails', { state: { countryData: item }});
    }

    renderCountryList = () => {
        return (
            <div className="countryContainer">
                <p className="searchedCountryText">Searched Country List</p>
                <div className="countryList">
                    {this.state.countryData.map((item: any, index: any) => {
                        if(!item)
                        return;
                        return (
                            <div onClick={() => this.onPressCountryDetails(item)} className="countryCell" key={index}>
                                <div>
                                    <p className="fieldValue">Country: {item?.name?.common}</p>
                                    {item?.capital && <p className="fieldValue">Capital: {item?.capital[0] ? item?.capital[0] : ''}</p>}
                                    <p className="fieldValue">Population: {item?.population}</p>
                                    <p className="fieldValue">Latlng: {item?.latlng[0]} {item?.latlng[1]}</p>
                                </div>
                                <img src={item.flags.svg} className="flagImage" />
                            </div>
                        )
                    })}
                </div>
            </div>
        )
    }

    render() {
        return (
            <div className='countryListMainContainer'>
                {this.renderCountryList()}
            </div>
        )
    }
};

export default WithRouter(CountryList);
export {CountryList};