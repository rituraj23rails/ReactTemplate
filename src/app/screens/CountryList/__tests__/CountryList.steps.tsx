import { defineFeature, loadFeature } from "jest-cucumber"
import { shallow, ShallowWrapper, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { CountryList } from "../CountryList";

configure({adapter: new Adapter()});

const screenProps = {
    navigate: jest.fn(),
    location: {
        state: {
            countryList: []
        }
    },
    id: "CountryList"
}

const feature = loadFeature('./src/app/screens/CountryList/__tests__/features/CountryList-scenarios.feature');

defineFeature(feature, (test) => {

    test('User navigates Country List Screen', ({ given, when, then }) => {
        let CountryListScreenWrapper: ShallowWrapper;
        let CountryListScreenInstance: CountryList;

        given('I am a User loading Country List Screen', () => {
            CountryListScreenWrapper = shallow(<CountryList {...screenProps} />)
        });

        when('I navigate to the Country List Screen', () => {
            CountryListScreenInstance = CountryListScreenWrapper.instance() as CountryList
        });

        then('Country List Screen will load with out errors', () => {
            CountryListScreenInstance.componentDidMount();
            expect(CountryListScreenWrapper).toBeTruthy()
            expect(CountryListScreenWrapper).toMatchSnapshot()
        });

        then('I can leave the screen with out errors', () => {
            CountryListScreenInstance.componentWillUnmount();
            expect(CountryListScreenWrapper).toBeTruthy()
            expect(CountryListScreenWrapper).toMatchSnapshot()
        });
    });

});
