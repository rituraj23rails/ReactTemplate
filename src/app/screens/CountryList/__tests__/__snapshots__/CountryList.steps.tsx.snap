// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country List Screen User navigates Country List Screen 1`] = `
<div
  className="countryListMainContainer"
>
  <div
    className="countryContainer"
  >
    <p
      className="searchedCountryText"
    >
      Searched Country List
    </p>
    <div
      className="countryList"
    />
  </div>
</div>
`;

exports[`Country List Screen User navigates Country List Screen 2`] = `
<div
  className="countryListMainContainer"
>
  <div
    className="countryContainer"
  >
    <p
      className="searchedCountryText"
    >
      Searched Country List
    </p>
    <div
      className="countryList"
    />
  </div>
</div>
`;
